streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -315,7 +314,7 @@ public void shouldCleanupStateDirectoriesWhenLastModifiedIsLessThanNowMinusClean
     }
 
     @Test
-    public void shouldCleanupObsoleteStateDirectoriesOnlyOnce() {
+    public void shouldCleanupObsoleteTaskDirectoriesAndDeleteTheDirectoryItself() {
```

Comment:
This test was originally added when we fixed a bug where Streams was continuously attempting to clean "empty" task directories and logging the `"Deleting obsolete state directory..."` message. The root cause of this was that we couldn't delete the task directory itself due to the lock file, so we just left empty directories behind. This PR fixes that, so I adapted the PR to test this new functionality since the old test wouldn't really make sense any more